engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/UpdateBy.java

```diff
@@ -552,56 +520,136 @@ public void close() {
          */
-        private void processWindowBucketOperators(int winIdx, int winOpIdx, final Runnable resumeAction) {
-            jobScheduler.iterateParallel(ExecutionContext.getContextToRecord(), this,
-                    JobScheduler.DEFAULT_CONTEXT_FACTORY,
+        private void processWindowBucketOperators(final int winIdx,
+                final int[] winOpArr,
+                final int maxAffectedChunkSize,
+                final int maxInfluencerChunkSize,
+                final Runnable onWindowBucketOperatorsComplete,
+                final Consumer<Exception> onWindowBucketOperatorError) {
+            final class OperatorThreadContext implements JobScheduler.JobThreadContext {
+                final Chunk<? extends Values>[] chunkArr;
+                final ChunkSource.GetContext[] chunkContexts;
+                final UpdateByOperator.Context[] winOpContexts;
+
+                OperatorThreadContext() {
+                    winOpContexts = new UpdateByOperator.Context[winOpArr.length];
+
+                    for (int ii = 0; ii < winOpArr.length; ii++) {
+                        final int opIdx = winOpArr[ii];
+                        winOpContexts[ii] = windows[winIdx].operators[opIdx].makeUpdateContext(maxAffectedChunkSize);
+                    }
+
+                    final int[] srcIndices = windows[winIdx].operatorInputSourceSlots[winOpArr[0]];
+                    chunkArr = new Chunk[srcIndices.length];
+                    chunkContexts = new ChunkSource.GetContext[srcIndices.length];
+
+                    for (int ii = 0; ii < srcIndices.length; ii++) {
+                        int srcIdx = srcIndices[ii];
+                        chunkContexts[ii] = maybeCachedInputSources[srcIdx].makeGetContext(maxInfluencerChunkSize);
+                    }
+                }
+
+                @Override
+                public void close() {
+                    SafeCloseableArray.close(winOpContexts);
+                    SafeCloseableArray.close(chunkContexts);
+                }
+            }
+
+            jobScheduler.iterateParallel(ExecutionContext.getContextToRecord(),
+                    chainAppendables(this, stringAndIndexToAppendable("-processWindowBucketOperators", winIdx)),
+                    OperatorThreadContext::new,
                     0, dirtyBuckets.length,
-                    (context, bucketIdx) -> {
+                    (context, bucketIdx, nec) -> {
                         UpdateByBucketHelper bucket = dirtyBuckets[bucketIdx];
-                        if (bucket.windowContexts[winIdx].isDirty
-                                && bucket.windowContexts[winIdx].dirtyOperators.get(winOpIdx)) {
-                            windows[winIdx].processBucketOperator(bucket.windowContexts[winIdx], winOpIdx, initialStep);
+                        if (bucket.windowContexts[winIdx].isDirty) {
+                            windows[winIdx].processBucketOperator(bucket.windowContexts[winIdx], winOpArr,
```

Comment:
`processBucketOperators`
Do make sure clean bucket-win-ops are filtered in the nested code.